uber/decorators.py

```diff
@@ -279,8 +274,72 @@ def __call__(self, klass):
 
 
 register = template.Library()
+
+
 def tag(klass):
     @register.tag(klass.__name__)
     def tagged(parser, token):
         return klass(*token.split_contents()[1:])
     return klass
+
+
+class Validation:
+    def __init__(self):
+        self.validations = defaultdict(OrderedDict)
+
+    def __getattr__(self, model_name):
+        def wrapper(func):
+            self.validations[model_name][func.__name__] = func
+            return func
+        return wrapper
+
+validation = Validation()
+
+
+adjustment_counter = count().__next__
+
+
```

Comment:
This is a new API for something we were doing already in an ad-hoc way.  I believe the docstrings explain this sufficiently, but let me know if they do not.